iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift

```diff
@@ -143,6 +143,12 @@ extension BMCDefaultViewModel {
     manager.shareCategory(category.categoryId)
   }
 
+  func shareAllCategories(handler: @escaping OnPreparedToShareHandler) {
+    // TODO: handle empty
+    onPreparedToShareCategory = handler
```

Comment:
Should this handler be passed as a parameter, or can it be just hard-coded? Isn't it always the same?

Reply:
No. This handler is stored as property and called later in the other method:

```swift
  func onBookmarksCategoryFilePrepared(_ status: BookmarksShareStatus) {
    switch status {
    case .success:
      onPreparedToShareCategory?(.success(manager.shareCategoryURL()))
    case .emptyCategory:
      onPreparedToShareCategory?(.error(title: L("bookmarks_error_title_share_empty"), text: L("bookmarks_error_message_share_empty")))
    case .archiveError: fallthrough
    case .fileError:
      onPreparedToShareCategory?(.error(title: L("dialog_routing_system_error"), text: L("bookmarks_error_message_share_general")))
    }
  }
```
which is callers from the BookmarksManager:
```
  [self loopObservers:^(id<MWMBookmarksObserver> observer) {
    if ([observer respondsToSelector:@selector(onBookmarksCategoryFilePrepared:)])
      [observer onBookmarksCategoryFilePrepared:status];
  }];
```

So it is a responsibility of the subscriber to handle and call this handler.

Reply:
Maybe I wasn't clear. Is shareAllCategories function accepting different handlers? I see only one. If it is only one, then why this handler can't be just a function? It should not be passed as a parameter here, and it should not be stored as a variable then.